core/spec/lib/spree/core/testing_support/factories/order_factory_spec.rb

```diff
@@ -45,4 +45,10 @@
 
     it_behaves_like 'a working factory'
   end
+
+  describe 'completed order with promotion' do
+    let(:factory) { :completed_order_with_promotion }
+
+    it_behaves_like 'a working factory'
+  end
```

Comment:
No, we only recently put the factories under test, and the coverage is not perfect yet. They're very complex, and we'll happily accept contributions towards improving both the factories and their specs. Yes, the test suite itself kinda sorta tests the factories, but that's all very implicit and meh. 

This change is good as far as I'm concerned, but it does not test the improvements to the factory. @graygilmore If you want to add specs that test your improvements, I'd be happy - but I don't see that as a blocker for an obvious improvement.